Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp

```diff
@@ -67,9 +67,15 @@ class TrackFindingAlgorithm final : public IAlgorithm {
   class TrackFinderFunction {
    public:
     virtual ~TrackFinderFunction() = default;
+    virtual TrackFinderResult operator()(
+        const TrackParameters& initialParameters,
+        const TrackFinderOptions& options, TrackContainer& tracks) const {
+      auto rootBranch = tracks.makeTrack();
+      return (*this)(initialParameters, options, tracks, rootBranch);
+    }
```

Comment:
Do we need this? Wouldn't it be better to have all calls do the `tracks.makeTrack()` in the caller. `TrackFindingAlgorithm.cpp` already does it, so it would just need to change `TrackFindingExaTrkX/src/TrackFindingFromPrototrackAlgorithm.cpp`. This would also save a second virtual function call.
I don't think we worry about compatibility in the Examples.